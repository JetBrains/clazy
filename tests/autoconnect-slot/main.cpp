#include <QtCore/QObject>
#include "ui_widget.h"

class MyObject : public QWidget, private Ui::Widget
{
    Q_OBJECT
public:
    MyObject()
    {
      setupUi( this );
    }
    bool on_mButton_something(); // No warn

public Q_SLOTS:
    void slot1() const {} // No warn
    void slot2_something() {} // No warn
    void slot3_something_(); // No warn
    void on_mButton_clicked(); // Warn
    void on_button_clicked(); // Warn
    void on_mButton_pressed(); // Warn
    void on_mButton_released() {} // Warn
    void on_mButton(); // No warn
    void on_notButton_clicked(); // No warn
    void on_mDouble_clicked(); // No warn - not a widget
    void on_mButton_updated(); // No warn - not a signal

  private:

    double mDouble;
    QPushButton* button;

};

void MyObject::on_mButton_pressed() // OK, already warned
{

}

void test()
{
    MyObject o;
}
