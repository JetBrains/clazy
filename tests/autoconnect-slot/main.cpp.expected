autoconnect-slot/main.cpp:18:5: warning: Use of autoconnected UI slot: on_mButton_clicked [-Wclazy-autoconnect-slot]
autoconnect-slot/main.cpp:19:5: warning: Use of autoconnected UI slot: on_button_clicked [-Wclazy-autoconnect-slot]
autoconnect-slot/main.cpp:20:5: warning: Use of autoconnected UI slot: on_mButton_pressed [-Wclazy-autoconnect-slot]
autoconnect-slot/main.cpp:21:5: warning: Use of autoconnected UI slot: on_mButton_released [-Wclazy-autoconnect-slot]
autoconnect-slot/main.cpp:33:1: warning: Use of autoconnected UI slot: on_mButton_pressed [-Wclazy-autoconnect-slot]
